tasktiger/worker.py

```diff
@@ -569,12 +570,30 @@ def check_child_exit():
 
                 # Wait until the timeout or a signal / child exit occurs.
                 try:
-                    select.select(
+                    result = select.select(
                         [pipe_r],
                         [],
                         [],
                         self.config['ACTIVE_TASK_UPDATE_TIMER'],
                     )
+
+                    if result[0]:
+                        # Purge pipe so select will pause on next call.
+                        # This is a blocking call but the select should
+                        # ensure at least one byte is available.
```

Comment:
Sounds like this may not be guaranteed.

> Under Linux, select() may report a socket file descriptor as "ready for reading", while nevertheless a subsequent read blocks. This could for example happen when data has arrived but upon examination has wrong checksum and is discarded. **There may be other circumstances in which a file descriptor is spuriously reported as ready.** Thus it may be safer to use O_NONBLOCK on sockets that should not block.

http://man7.org/linux/man-pages/man2/select.2.html